pkg/util/chmod/chmod.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package chmod
+
+import (
+	"os"
+)
+
+type Interface interface {
+	Chmod(path string, filemode os.FileMode) error
+}
+
+func New() Interface {
+	return &chmodRunner{}
+}
+
+type chmodRunner struct{}
+
+func (_ *chmodRunner) Chmod(path string, mode os.FileMode) error {
+	return os.Chmod(path, mode)
```

Comment:
It's meant to, but it's not.  I think the Golang way here is to use it in conjunction with `path/filepath` to walk the tree.  I will change, good catch.